src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/DataverseXoaiItemRepository.java

```diff
@@ -0,0 +1,329 @@
+        // should be available for this record. 
+        
+        List<OAIRecord> oaiRecords = recordService.findOaiRecordsByGlobalId(identifier);
+        if (oaiRecords != null && !oaiRecords.isEmpty()) {
+            DataverseXoaiItem xoaiItem = null; 
+            for (OAIRecord oaiRecord : oaiRecords) {
+                // We can return the first *active* record we find for this identifier. 
+                if (!oaiRecord.isRemoved()) {
+                    xoaiItem = new DataverseXoaiItem(oaiRecord);
+                    return xoaiItem; 
+                }
+            }
+        }
+        
+        throw new IdDoesNotExistException();
+    }
+    
+    @Override
+    public Item getItem(String identifier, MetadataFormat metadataFormat) throws HandlerException {
+        logger.fine("getItem; calling findOaiRecordsByGlobalId, identifier " + identifier);
+        
+        if (metadataFormat == null) {
+            throw new NoMetadataFormatsException("Metadata Format is Required");
+        }
+        
+        List<OAIRecord> oaiRecords = recordService.findOaiRecordsByGlobalId(identifier);
+        if (oaiRecords != null && !oaiRecords.isEmpty()) {
+            DataverseXoaiItem xoaiItem = null; 
+            for (OAIRecord oaiRecord : oaiRecords) {
+                if (xoaiItem == null) {
+                    xoaiItem = new DataverseXoaiItem(oaiRecord); 
+                    
+                    // If this is a "deleted" OAI oaiRecord - i.e., if someone
+                    // has called GetRecord on a deleted oaiRecord (??), our 
+                    // job here is done. If it's a live oaiRecord, let's try to 
+                    // look up the dataset and open the pre-generated metadata 
+                    // stream. 
+                    
+                    if (!oaiRecord.isRemoved()) {
+                        Dataset dataset = datasetService.findByGlobalId(oaiRecord.getGlobalId());
+                        if (dataset == null) {
+                            // This should not happen - but if there are no longer datasets 
+                            // associated with this persistent identifier, we should simply 
+                            // bail out. 
+                            // TODO: Consider an alternative - instead of throwing 
+                            // an IdDoesNotExist exception, mark the oaiRecord as 
+                            // "deleted" and serve it to the client (?). For all practical
+                            // purposes, this is what this oaiRecord represents - it's 
+                            // still in the database as part of an OAI set; but the 
+                            // corresponding dataset no longer exists, because it 
+                            // must have been deleted. 
+                            // i.e.
+                            // xoaiItem.getOaiRecord().setRemoved(true);
+                            break;
```

Comment:
**suggestion:** Why not simple throw the exception here to make it more explicit and reduce the danger of missing this when refactoring? (Is there a unit test for the behavior?)